src/main/java/mate/academy/bookstoreappspring/service/BookServiceImpl.java

```diff
@@ -55,4 +59,13 @@ public BookDto updateById(Long id, UpdateBookRequestDto updateBookRequestDto) {
         Book updatedBook = bookRepository.save(book);
         return bookMapper.toDto(updatedBook);
     }
+
+    @Override
+    public List<BookDto> search(BookSearchParamsDto bookSearchParams) {
+        Specification<Book> spec = bookSpecificationBuilder.build(bookSearchParams);
+
+        List<Book> all = bookRepository.findAll(spec);
+
+        return all.stream().map(bookMapper::toDto).toList();

```

Comment:
There is no need to create this temporary variable

```suggestion
        return bookRepository.findAll(bookSpecification).stream()
                .map(bookMapper::toDto)
                .toList();

```